### PR TITLE
Support for manage tag

### PR DESCRIPTION
**Pull request by jayanthvn**
**What type of PR is this?**
Enhancement

**Which issue does this PR fix**:
Certain use cases need to use only specific ENIs for pod IPs similar to allow and deny list. Currently we support no_manage tag but certain use cases need both manage and no_manage tag.

**What does this PR do / Why do we need it**:
With this knob, only ENIs marked with `node.k8s.amazonaws.com/manage` tag, primary ENI and previously created CNI created ENI attached to the instance will be used for pod IP allocation.

Name of the knob is still under review.

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:
N/A

**Testing done on this change**:

Yes, UTs will be taken up in the follow up PR

Manage tagged ENI -
```
{"level":"info","ts":"2021-09-10T17:40:22.414Z","caller":"ipamd/ipamd.go:934","msg":"Adding 192.168.10.46/32 to DS for eni-0d72a77d6d4efc1b9"}
{"level":"info","ts":"2021-09-10T17:40:22.414Z","caller":"ipamd/ipamd.go:934","msg":"Added ENI(eni-0d72a77d6d4efc1b9)'s IP/Prefix 192.168.10.46/32 to datastore"}
{"level":"info","ts":"2021-09-10T17:40:22.414Z","caller":"ipamd/ipamd.go:934","msg":"Adding 192.168.22.174/32 to DS for eni-0d72a77d6d4efc1b9"}
{"level":"info","ts":"2021-09-10T17:40:22.414Z","caller":"ipamd/ipamd.go:934","msg":"Added ENI(eni-0d72a77d6d4efc1b9)'s IP/Prefix 192.168.22.174/32 to datastore"}
{"level":"info","ts":"2021-09-10T17:40:22.414Z","caller":"ipamd/ipamd.go:934","msg":"Adding 192.168.4.160/32 to DS for eni-0d72a77d6d4efc1b9"}
{"level":"info","ts":"2021-09-10T17:40:22.414Z","caller":"ipamd/ipamd.go:934","msg":"Added ENI(eni-0d72a77d6d4efc1b9)'s IP/Prefix 192.168.4.160/32 to datastore"}
{"level":"info","ts":"2021-09-10T17:40:22.414Z","caller":"ipamd/ipamd.go:934","msg":"Adding 192.168.22.102/32 to DS for eni-0d72a77d6d4efc1b9"}
{"level":"info","ts":"2021-09-10T17:40:22.414Z","caller":"ipamd/ipamd.go:934","msg":"Added ENI(eni-0d72a77d6d4efc1b9)'s IP/Prefix 192.168.22.102/32 to datastore"}
{"level":"info","ts":"2021-09-10T17:40:22.414Z","caller":"ipamd/ipamd.go:934","msg":"Adding 192.168.29.135/32 to DS for eni-0d72a77d6d4efc1b9"}
{"level":"info","ts":"2021-09-10T17:40:22.414Z","caller":"ipamd/ipamd.go:934","msg":"Added ENI(eni-0d72a77d6d4efc1b9)'s IP/Prefix 192.168.29.135/32 to datastore"}
{"level":"debug","ts":"2021-09-10T17:40:22.414Z","caller":"ipamd/ipamd.go:857","msg":"Datastore Pool stats: total(/32): 5, assigned(/32): 0, total prefixes(/28): 0"}
```
No Manage tagged ENI- 

```
{"level":"info","ts":"2021-09-10T17:41:22.586Z","caller":"ipamd/ipamd.go:1153","msg":"eni-0d72a77d6d4efc1b9 is of type: interface"}
{"level":"debug","ts":"2021-09-10T17:41:22.586Z","caller":"ipamd/ipamd.go:1173","msg":"Marking ENI eni-05b07024baab477b5 as being unmanaged"}
{"level":"debug","ts":"2021-09-10T17:41:22.586Z","caller":"ipamd/ipamd.go:1175","msg":"Skipping ENI eni-05b07024baab477b5: tagged with node.k8s.amazonaws.com/no_manage"}
```
**Automation added to e2e**:

Yes will add in follow up PR

**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
No

**Does this change require updates to the CNI daemonset config files to work?**:

No

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -442,7 +442,10 @@ The label notifies vpc-resource-controller (https://github.com/aws/amazon-vpc-re
 Once enabled the VPC resource controller will then advertise branch network interfaces as extended resources on these nodes in your cluster. Branch interface capacity is additive to existing instance type limits for secondary IP addresses and prefixes. For example, a c5.4xlarge can continue to have up to 234 secondary IP addresses or 234 /28 prefixes assigned to standard network interfaces and up to 54 branch network interfaces. Each branch network interface only receives a single primary IP address and this IP address will be allocated to pods with security group(branch ENI pods).
 
 Any of the WARM targets do not impact the scale of the branch ENI pods so you will have to set the WARM_{ENI/IP/PREFIX}_TARGET based on the number of non-branch ENI pods. If you are having the cluster mostly using pods with security group consider setting WARM_IP_TARGET to a very low value instead of default WARM_ENI_TARGET or WARM_PREFIX_TARGET to reduce wastage of IPs/ENIs.
+<<<<<<< HEAD
 
+=======
```
Comment (M00nF1sh) [inline README.md:447]: seems merge conflicts

File: config/master/aws-k8s-cni-cn.yaml [hunk 2]
```diff
@@ -156,6 +156,8 @@
           "value": "false"
         - "name": "ENABLE_PREFIX_DELEGATION"
           "value": "false"
+        - "name": "ENI_MANAGE_MODE"
```
Comment (M00nF1sh) [inline config/master/aws-k8s-cni-cn.yaml:159]: i feel we don't need to add all env to our manifest if it have a default value in code

File: pkg/awsutils/vpc_ip_resource_limit.go [hunk 3]
```diff
@@ -19,420 +19,420 @@ package awsutils
 // InstanceNetworkingLimits contains a mapping from instance type to networking limits for the type. Documentation found at
 // https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html#AvailableIpPerENI
 var InstanceNetworkingLimits = map[string]InstanceTypeLimits{
-	"a1.2xlarge":    {ENILimit: 4, IPv4Limit: 15, HypervisorType:"nitro"},
```
Comment (M00nF1sh) [inline pkg/awsutils/vpc_ip_resource_limit.go:22]: why is this file changed

File: README.md [hunk 4]
```diff
@@ -505,6 +508,17 @@ Setting `DISABLE_NETWORK_RESOURCE_PROVISIONING` to `true` will make IPAMD to dep
 
 ---
 
+#### `ENI_MANAGE_MODE` (v1.9.1+)
```
Comment (M00nF1sh) [inline README.md:511]: this knob should actually control what an "Untagged" ENI should be considered as managed or unmanaged.

* if ENI have `node.k8s.amazonaws.com/no_manage`, it should always be considered as no_managed.
* if ENI have `node.k8s.amazonaws.com/manage`, it should always be considered as managed.
* if ENI is created by IPAMD(have `node.k8s.amazonaws.com/instance_id`), it should always be considered as managed.
* 
* If ENI don't fall into any category above, this knobs determines whether it's managed or unmanaged.

File: README.md (hunk 1 above)
Comment (jayanthvn) [reply README.md:447]: Oh somehow it creeped in :D

File: config/master/aws-k8s-cni-cn.yaml (hunk 2 above)
Comment (jayanthvn) [reply config/master/aws-k8s-cni-cn.yaml:159]: Yeah but generally we add it so that it is clearer for the reader.

File: pkg/awsutils/vpc_ip_resource_limit.go (hunk 3 above)
Comment (jayanthvn) [reply pkg/awsutils/vpc_ip_resource_limit.go:22]: make format, did this. Let me revert this file.

File: README.md (hunk 4 above)
Comment (jayanthvn) [reply README.md:511]: yeah sure, I can add something like this.

**Comment by jayanthvn** [comment]
Will create a new PR.
